validate.py

```diff
@@ -1,111 +1,125 @@
+# Copyright (C) 2019 University of Michigan ITS Teaching and Learning
 
-UNIZIN_FILE_FORMAT = "unizin_{table}.csv"
+# Local modules
+from dbqueries import QUERIES
 
-## don't modify anything below this line (except for experimenting)
-
-import sys, os, csv, itertools, argparse, smtplib, tempfile, json
+# Standard modules
+import json, logging, os, smtplib, sys
+from datetime import datetime
 
+# Third-party modules
 from email.mime.text import MIMEText
-from collections import OrderedDict 
-
-from dateutil import parser as dparser
-
-import psycopg2
-
-import numpy as np
 import pandas as pd
+import psycopg2, pytz
 
-import dbqueries
-
-from collections import OrderedDict
-from operator import itemgetter
-
-from tqdm import tqdm
+# Global variables
 
+logging.basicConfig(level="DEBUG")
+logger = logging.getLogger(__name__)
 
 try:
-    with open(os.getenv("ENV_FILE", "/unizin-csv-validation/config/env.json")) as f:
+    with open(os.getenv("ENV_FILE", "config/env.json")) as f:
         ENV = json.load(f)
 except FileNotFoundError as fnfe:
-    print("Default config file or one defined in environment variable ENV_FILE not found. This is normal for the build, should define for operation")
+    logger.debug("Default config file or one defined in environment variable ENV_FILE not found. This is normal for the build, should define for operation.")
     # Set ENV so collectstatic will still run in the build
     ENV = os.environ
 
-OUT_DIR = ENV.get("TMP_DIR", "/tmp/")
-RESULTS_FILE = open(OUT_DIR + "u_results.txt", "w")
-ERRORS_FILE = open(OUT_DIR + "u_errors.txt", "w")
-
-def load_unizin_to_csv(tablename):
-    out_filename = OUT_DIR + UNIZIN_FILE_FORMAT.format(table=tablename)
-    print (f"Loading {tablename} table to {out_filename}")
-    # The DSN might switch depending on the data file
-    conn = psycopg2.connect(ENV.get("DSN_"+dbqueries.QUERIES[tablename]['dsn']))
-    
-    curs = conn.cursor()
-
-    query = dbqueries.QUERIES[tablename]
-    if (query.get('prequery')):
-        curs.execute(query.get('prequery'))
-    UWriter = open(out_filename,"w")
-    # Try this first with this, breaks in 8.0 though :()
-    try:
-        outputquery = "COPY ({0}) TO STDOUT WITH CSV HEADER FORCE QUOTE *".format(query.get('query'))
-        curs.copy_expert(outputquery, UWriter)
-    except psycopg2.ProgrammingError:
-        print ("Copy query failed, trying regular query, possibly Postgres 8.0")
-        writer = csv.writer(UWriter)
-
-        conn.reset()
-        curs.execute(query.get('query'))
-        writer.writerows(curs.fetchall())
-
-def email_results(filenames, subject=None):
-    #Email results, all filenames should be csv with key/value pairs
+OUT_DIR = ENV.get("OUT_DIR", "data/")
+
+
+# Functions
+
+def establish_db_connection(db_name):
+    conn_params = ENV.get(db_name)
+    if conn_params['type'] == "PostgreSQL":
+        conn = psycopg2.connect(
+            host=conn_params["host"],
```

Comment:
I **think** if you changed the name value to dbname this could just be simplified to be just since the other values map straight. 

`connect(**conn_params)`

Reply:
That makes sense. I'd have to move the `type` key/value pair elsewhere, but that would be okay.

Reply:
Done. See latest [commit](https://github.com/tl-its-umich-edu/unizin-validation/pull/7/commits/f7b1096af1e237e424d0306fb54c6503794c0fca).